google/stackdriver_monitoring/stackdriver_handlers.py

```diff
@@ -98,78 +187,206 @@ def __init__(self, options, stackdriver):
-    if accepts_html(request):
-      html_rows = [('<tr><td>{0}</td><td>{1}</td></tr>'
-                    .format(all_names[i], handler.batch_response[i]))
-                   for i in range(len(all_names))]
-      html_body = 'Deleted {0} of {1}:\n<table>\n{2}\n</table>'.format(
-          handler.num_ok, len(all_names), '\n'.join(html_rows))
-      html_doc = request.build_html_document(html_body, title='Cleared Time Series')
-      request.respond(response_code, {'ContentType': 'text/html'}, html_doc)
-    else:
-      text = ['{0}  {1}'.format(all_names[i], handler.batch_response[i])
-              for i in range(len(all_names))]
-      text.append('')
-      text.append('Deleted {0} of {1}'.format(
-            handler.num_ok, len(all_names)))
-      request.respond(response_code,
-                      {'ContentType': 'text/plain'},
-                      '\n'.join(text))
+    for index, ok in enumerate(delete_processor.was_ok):
+      if ok:
+        success_list.append(delete_list[index])
+      else:
+        failed_list.append(delete_list[index])
+        failed_errors.append(delete_processor.batch_response[index])
+
+  def __do_batch_update_create_helper(
+      self, project, create_list, success_list, failed_list, failed_errors):
+    get_descriptor_name = lambda descriptor: descriptor['name']
+    create_method = (self.__stackdriver.service.projects()
+                     .metricDescriptors().create)
+    def create_invocation(descriptor):
+      name = descriptor['name']
+      logging.info('batch CREATE %s', name)
+      return create_method(
+          name='projects/{0}'.format(project), body=descriptor)
+
+    create_processor = BatchProcessor(
+        project, self.__stackdriver,
+        create_list, create_invocation, get_descriptor_name)
+    create_processor.process()
+
+    for index, ok in enumerate(create_processor.was_ok):
+      if ok:
+        success_list.append(create_list[index])
+      else:
+        failed_list.append(create_list[index])
+        failed_errors.append(create_processor.batch_response[index])
+
+  def __do_batch_update(self, project, update_list, original_type_map):
+    get_descriptor_name = lambda descriptor: descriptor['name']
+
+    delete_errors = []
+    create_errors = []
+    restore_errors = []
+
+    failed_list = []
+    create_list = []
+    success_list = []
+    restore_list = []
+    not_updated_list = []
+    lost_list = []
+
+    if update_list:
+      self.__do_batch_update_delete_helper(
+          project, update_list, create_list, failed_list, delete_errors)
+
+    if create_list:
+      self.__do_batch_update_create_helper(
+        project, create_list, success_list, restore_list, create_errors)
+      restore_list = [original_type_map[elem['type']] for elem in restore_list]
+
+    if restore_list:
+      self.__do_batch_update_create_helper(
+        project, restore_list, not_updated_list, lost_list, restore_errors)
```

Comment:
Why is `not_updated_list` the `success` parameter here? I'm just asking from a semantic perspective to understand what the intent is.